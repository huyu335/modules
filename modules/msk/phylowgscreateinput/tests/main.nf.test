nextflow_process {

    name "Test Process PHYLOWGSCREATEINPUT"
    script "../main.nf"
    process "PHYLOWGSCREATEINPUT"

    tag "modules"
    tag "modules_nfcore"
    tag "phylowgscreateinput"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("PHYLOWGSCREATEINPUT - txt") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['cnvs_txt'], checkIfExists: true)
                    ]
                input[1] =  [ 
                    file(params.test_data_mskcc['neoantigen']['unfilteredmaf'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
