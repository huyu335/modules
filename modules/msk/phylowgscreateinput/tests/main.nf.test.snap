{
    "PHYLOWGSCREATEINPUT - txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,9228c0a7bce478e06db5d4304be3dbf7",
                        "ssm_data.txt:md5,055b7b5359a50eefd86fc85fcc023c6f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,71f53efef22d06969b865b3034a8bf67"
                ],
                "phylowgsinput": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,9228c0a7bce478e06db5d4304be3dbf7",
                        "ssm_data.txt:md5,055b7b5359a50eefd86fc85fcc023c6f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,71f53efef22d06969b865b3034a8bf67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-11T12:23:27.528815"
    }
}